tutorials/chapter1_annexB.md

```diff
@@ -0,0 +1,322 @@
+        (none)
+```
+
+**Threads.**
+
+Same 3 threads than in the previous example: main, publisher, an listener.
+
+**TestEvent constructors.**
+
+When the publisher are created:
+
+- 8x `emptyConstructorInvocations`: the smart memory pool are populated at the selected size.
+
+Just after the publishing:
+
+- same 3x `constructorInvocations` and 3x `destructorInvocations` but,
+- 0x `copyInvocations`, because listener data was aready allocated at the 8x pool.
+Go to example B5 and B6 to study a smaller pool.
+
+When the listeners finish,
+
+- `destructorInvocations` keeps the same, only the pool needs to be freed.
+
+<p align="center">
+constructorInvocations = destructorInvocations
+</p>
+
+## Example B3: Adding an initializer
+
+One step further example B2:
+
+```diff
+- const auto initializerFunction{nullptr};
++ const auto initializerFunction{[](TestEvent &event) { event.init(); }};
+```
+
+```C++
+    kpsrPublisher = eventLoop.getPublisher<TestEvent>("TestEvent",
+                                                      poolSize,
+                                                      initializerFunction,
+                                                      eventClonerFunction);
+```
+
+The `init()` function allocates the required object memory. As explained in [Smart Memory Pool](./chapter1.md#smart-memory-pool), this prevents the memory allocation on every publish event, increasing the application performance.
+
+### Sample output
+
+Differences with example B2:
+
+```text
+$ bin/kpsr_tutorial_chapter1_annexB_3
+
+Invocations counters updating after getPublisher
+        emptyConstructorInvocations 8
+        initInvocations             8
+```
+
+**TestEvent constructors.**
+
+When the publisher are created:
+
+- 8x `emptyConstructorInvocations`: the smart memory pool are populated as before and,
+- 8x `initInvocations`, every object is initialized to allocate its required memory.
+
+## Example B4: Adding a cloner
+
+One additional step from example B3:
+
+```diff
+- const auto eventClonerFunction{nullptr};
++ const auto eventClonerFunction{[](const TestEvent &src, TestEvent &dst) { dst = TestEvent(src); }};
+```
+
+```C++
+    kpsrPublisher = eventLoop.getPublisher<TestEvent>("TestEvent",
+                                                      poolSize,
+                                                      initializerFunction,
+                                                      eventClonerFunction);
+```
+
+Klepsydra, by default, copy the publisher objects to the subscriber with the equal (=) operator. This makes a copy _by value_ for simple objects an vectors, but not for heavy or complex data structures such as images or `cv::Mat`.\
+If your concurrent application could update the data while the listener is running, provide a cloner function to prevent undesired effects. The copy constructor at `include/test_event.h` performs this operation (although not required for this simple object).
```

Comment:
```suggestion
If the data you are publishing needs a different operation for deep copies, provide a cloner function to prevent undesired effects. In this example, the copy constructor at `include/test_event.h` performs this operation (although not required for this simple object).
```